### ISSUE TITLE
bug caused by shelljs modifying the String.prototype 

### ISSUE THREAD

**Issue by jonschlinkert** [eslint/eslint#7316]
* **ESLint Version:** 3.6.1
* **Node Version:** 6.7
* **npm Version:**: 2.15.8

tldr: I've confirmed that this bug is fixed by updating to the latest version of shelljs.

**What parser (default, Babel-ESLint, etc.) are you using?**

I'm using gulp-eslint, but the bug is caused by shelljs 0.6.0, a dependency of eslint. 

**What's happening**

I don't know why this is happening, but I can explain what is happening and what is causing it. 

When I run `gulp mocha`, even though `gulp eslint` is never called, since shelljs v6.0.0 modifies the `String.prototype` AND it's exported at the top level (e.g. regardless of whether or not shelljs is used, the `String.prototype` is modified), something is happening that is causing test results to be mutated. 

As you can see in the [attached gist](https://gist.github.com/jonschlinkert/528fe3ca9a6d0d2b64200e90c233aae2), only tests with a `..` in the expected result are being modified. It appears that `..` is being replaced with function from shelljs, so `.toString()` is probably being called somewhere.

I don't know where the replacement is happening specifically, but I tracked the root cause to [this code in shelljs](https://github.com/shelljs/shelljs/blob/fe06baf1173ec6f0a70cd58ddb7d373f4c6446f5/shell.js#L56-L62).  

Also, for completeness, the `mocha` task in `gulpfile.js` consists of this code:

```js
gulp.task('test', function() {
  return gulp.src('test/*.js')
    .pipe(mocha({reporter: 'spec'}))
});
```

**Comment by platinumazure** [comment]
Just to be sure, if you remove the gulp-eslint task and ensure ESLint is not loaded, you definitely don't see the issue anymore?

**Comment by platinumazure** [comment]
Oh, and now I just saw that you said latest ShellJS fixes this.

Feel free to submit a PR if you like, otherwise we can try to upgrade at some point.

**Comment by jonschlinkert** [comment]
> Just to be sure, if you remove the gulp-eslint task and ensure ESLint is not loaded, you definitely don't see the issue anymore?

yes that's correct.

> Feel free to submit a PR if you like, otherwise we can try to upgrade at some point.

Great, I'll submit one in the next couple of days, unless someone beats me to it. thanks

**Comment by platinumazure** [comment]
Also: I do find it weird that shelljs is being loaded at all. I'm fairly sure we only use it in our internal Makefile.js, which shouldn't be loaded in normal linting flows. So I'll take a look at some point and try to find if we require it elsewhere or if maybe gulp-eslint is pulling that file in.

**Comment by not-an-aardvark** [comment]
If I put `console.log(''.to)` in a rule and run ESLint's tests, it gets identified as a function. So I think the problem is with ESLint and not with gulp-eslint.

**Comment by mysticatea** [comment]
Thank you for this issue.

Ah, yes, ESLint imports `shelljs` to use `shelljs.test()` and `shelljs.exec()`.
It's not good if `shalljs` modifies the prototypes of built-in objects.

**Comment by jonschlinkert** [comment]
Hmm, I tried updating and it caused tests to fail. I probably won't have time to investigate further, in case someone else wants to look into it.

**Comment by gyandeeps** [comment]
I am working on this. There was a accidental breaking change introduced in shellljs. Which I have fixed now. Will open a PR once shelljs makes a new release.
Ref: https://github.com/shelljs/shelljs/pull/530

**Comment by jonschlinkert** [comment]
thanks!
